src/OpenFeature/EventExecutor.cs

```diff
@@ -10,7 +10,9 @@
 
 namespace OpenFeature
 {
-    internal sealed partial class EventExecutor : IAsyncDisposable
+    internal delegate Task ShutdownDelegate(CancellationToken cancellationToken);
```

Comment:
`IAsyncDisposable` was removed. I would encourage you to keep it since disposing of it will trigger the provider's shutdown.

Reply:
It didn't seem to me that there was simple way of leveraging IDisposableAsync with a cancellation token, so I removed this on favor of just shutdown so we can propagate the token. 

This is an internal class only used by us in a single location, so I didn't think that ability to use "using" for cleanup was that valuable.

Reply:
Relevant: https://github.com/open-feature/dotnet-sdk/pull/268#discussion_r1587609537

Reply:
I prefer to keep the `Dispose`.

I would do something like this:
```
public ValueTask DisposeAsync() => new(this.Shutdown(CancellationToken.None));
```

When disposing of the objects, we shouldn't have any cancellation tokens. 

What are your thoughts?

Reply:
> public ValueTask DisposeAsync() => new(this.Shutdown(CancellationToken.None));

Yes, that works syntactically, but it doesn't propagate the cancellation token from the original method the user called (`.ShutdownAsync`) so there's no point to involving the cancellation at all. In this case, we might as well remove the cancellation token entirely from this part of the API: https://github.com/open-feature/dotnet-sdk/pull/268#issuecomment-2090536154

Reply:
I've removed the cancellation and made this `IAsyncDisposable` again.